src/depthai_wrapper.cpp

```diff
@@ -0,0 +1,169 @@
+#include "depthai/depthai_wrapper.hpp"
+
+namespace DepthAI
+{
+
+DepthAI::DepthAI(std::string usb_device, std::string config_file, bool usb2_mode) :
+      Device(usb_device, usb2_mode)
+{
+
+    std::ifstream file(config_file);
+    std::ostringstream file_stream;
+    file_stream << file.rdbuf();
+    std::string config_str = file_stream.str();
+
+    pipeline_ = this->create_pipeline(config_str);
+    config_json_ = nlohmann::json::parse(config_str);
+
+    this->set_resolution();
+    this->create_frame_holders();
+
+    width_map_[720] = 1280;
+    width_map_[800] = 1280;
+    width_map_[400] = 640;
```

Comment:
Are these always fixed? Or should we put them in a file and load them?

If they're always constant, you should create a `static const std::unordered_map<T1, T2>` class object for this file instead of doing this inside the constructor.